src/gt4py/next/ffront/past_to_itir.py

```diff
@@ -141,16 +142,42 @@ def _visit_stencil_call(self, node: past.Call, **kwargs) -> itir.StencilClosure:
 
         assert isinstance(node.func.type, (ts_ffront.FieldOperatorType, ts_ffront.ScanOperatorType))
 
-        lowered_args, lowered_kwargs = type_info.canonicalize_arguments(
+        args, node_kwargs = type_info.canonicalize_arguments(
             node.func.type,
-            self.visit(node.args, **kwargs),
-            self.visit(node_kwargs, **kwargs),
+            node.args,
+            node_kwargs,
             use_signature_ordering=True,
         )
 
+        lowered_args, lowered_kwargs = self.visit(args, **kwargs), self.visit(node_kwargs, **kwargs)
+
+        stencil_params = []
+        stencil_args = []
+        for i, arg in enumerate([*args, *node_kwargs]):
+            stencil_params.append(f"__stencil_arg{i}")
+            if isinstance(arg.type, ts.TupleType):
+                # convert into tuple of iterators
+                stencil_args.append(
+                    lowering_utils.to_tuples_of_iterator(f"__stencil_arg{i}", arg.type)
+                )
+            else:
+                stencil_args.append(f"__stencil_arg{i}")
+
+        if isinstance(node.func.type, ts_ffront.ScanOperatorType):
+            # scan operators return an iterator of tuples
+            stencil_body = im.deref(im.call(node.func.id)(*stencil_args))
+        else:
+            # field operators return a tuple of iterators
+            # TODO(tehrengruber): change into deref per element
+            stencil_body = im.deref(
+                lowering_utils.to_iterator_of_tuples(
+                    im.call(node.func.id)(*stencil_args), node.func.type.definition.returns
+                )
+            )
```

Comment:
why not implement the todo? seems like all utils are there?

Reply:
True I decided to keep effort low (and honestly also motivation as I am in this continuous mini-change, retest cycle) as these functions will become superfluous with combined IR. If you would like to have it I'll do it.